### PR TITLE
feat: add autoplay preference

### LINKED ISSUE #650: [Feature]: Add auto play for tv series
### What feature do you want to add?

add auto play to the on screen ui so you can toggle on or off so you dont have to go and choose the next episode.

### Why do you want to have this feature?

possibly just a tab on/off switch in the episodes menu

### Anything other details to share?

_No response_

### PR DESCRIPTION
**Pull request by qtchaos**
This pull request resolves #650. 

~~Still need to fix the lodash mess and properly debounce/throttle or whatever the next episode.~~

 - [X] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [X] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [X] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [X] I have tested all of my changes.

### CODE REVIEW

File: src/pages/parts/settings/PreferencesPart.tsx [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
 import { Toggle } from "@/components/buttons/Toggle";
 import { FlagIcon } from "@/components/FlagIcon";
 import { Dropdown } from "@/components/form/Dropdown";
 import { Heading1 } from "@/components/utils/Text";
+import { conf } from "@/setup/config";
 import { appLanguageOptions } from "@/setup/i18n";
+import { useAuthStore } from "@/stores/auth";
 import { getLocaleInfo, sortLangCodes } from "@/utils/language";
 
+function useIsExtensionActive() {
```
Comment (JorrinKievit) [inline src/pages/parts/settings/PreferencesPart.tsx:14]: Why can't you use `isExtensionActiveCached()`?

File: src/components/player/atoms/NextEpisodeButton.tsx [hunk 2]
```diff
@@ -95,6 +100,25 @@ export function NextEpisodeButton(props: {
...
 
+  useEffect(() => {
+    if (!enableAutoplay || metaType !== "show") return;
+
+    const interval = setInterval(() => {
+      const onePercent = duration / 100;
+      const isEnding = time >= duration - onePercent && duration !== 0;
+
+      const allowAutoplay = Boolean(
+        conf().ALLOW_AUTOPLAY ||
```
Comment (JorrinKievit) [inline src/components/player/atoms/NextEpisodeButton.tsx:111]: Can we extract this to a function in some util file with a descriptive name? I saw this used somewhere else

File: src/components/player/atoms/NextEpisodeButton.tsx [hunk 3]
```diff
@@ -95,6 +100,25 @@ export function NextEpisodeButton(props: {
     updateItem,
   ]);
 
+  useEffect(() => {
+    if (!enableAutoplay || metaType !== "show") return;
+
+    const interval = setInterval(() => {
```
Comment (JorrinKievit) [inline src/components/player/atoms/NextEpisodeButton.tsx:106]: Could you explain why we need an interval here, and not just wait for the show to end?

File: src/components/player/atoms/NextEpisodeButton.tsx [hunk 4]
```diff
@@ -95,6 +100,25 @@ export function NextEpisodeButton(props: {
...
+      const onePercent = duration / 100;
+      const isEnding = time >= duration - onePercent && duration !== 0;
+
+      const allowAutoplay = Boolean(
+        conf().ALLOW_AUTOPLAY ||
+          isExtensionActiveCached() ||
+          useAuthStore.getState().proxySet,
```
Comment (JorrinKievit) [inline src/components/player/atoms/NextEpisodeButton.tsx:113]: I think it should be `(ALLOW_AUTOPLAY && proxySet) || isExtensionActiveChached()` right?

File: src/pages/parts/settings/PreferencesPart.tsx [hunk 5]
```diff
@@ -62,6 +88,32 @@ export function PreferencesPart(props: {
...
+        <p className="max-w-[25rem] font-medium">
+          {t("settings.preferences.autoplayDescription")}
+        </p>
+        <div
+          onClick={() =>
+            allowAutoplay
+              ? props.setEnableAutoplay(!props.enableAutoplay)
+              : null
+          }
+          className="bg-dropdown-background hover:bg-dropdown-hoverBackground select-none my-4 cursor-pointer space-x-3 flex items-center max-w-[25rem] py-3 px-4 rounded-lg"
```
Comment (JorrinKievit) [inline src/pages/parts/settings/PreferencesPart.tsx:104]: Think you can use the classNames function here to apply the conditional styles below

**Review by JorrinKievit — CHANGES_REQUESTED** [review]

File: src/components/player/atoms/NextEpisodeButton.tsx (hunk 4 above)
Comment (qtchaos) [reply src/components/player/atoms/NextEpisodeButton.tsx:113]: 1. If autoplay is enabled by the host, then you can use the default proxy. 
2. if a custom proxy is set, then you can use
3. if the extension is set, you can use it.

Comment (JorrinKievit) [reply src/components/player/atoms/NextEpisodeButton.tsx:113]: Aaah true nevermind, makes sense

File: src/components/player/atoms/NextEpisodeButton.tsx (hunk 3 above)
Comment (qtchaos) [reply src/components/player/atoms/NextEpisodeButton.tsx:106]: If the show ends without an interval, it skips to the very last episode of the season.

Comment (JorrinKievit) [reply src/components/player/atoms/NextEpisodeButton.tsx:106]: I'm not sure I understand, why does it skip to the last episode?

Comment (qtchaos) [reply src/components/player/atoms/NextEpisodeButton.tsx:106]: I think its a dependency loop (if thats even a thing). Once the `loadNextEpisode` function gets called it updates and so due it being a dependency it creates a loop?

**Review by JorrinKievit — APPROVED** [review]
Works great!

**Comment by JorrinKievit** [comment]
Let's not forget to update the docs with the new env variable

**Pull request merged**